### ISSUE TITLE
JS ERROR: can't convert "y" to an integer

### ISSUE THREAD

**Issue by N-Coder** [micheleg/dash-to-dock#188]
After upgrading to GNOME Shell 3.16.2 I get a lot of errors in the journalctl log stating that "y" can't be converted to an integer. I'm using Dash to Dock version 46 on fedora 22.
~~~~~~
Jun 28 17:42:27 gnome-session[16773]: (gnome-shell:16909): Gjs-WARNING **: JS ERROR: Error: can't convert "y" to an integer
Jun 28 17:42:27 gnome-session[16773]: myAppIcon<.updateIconGeometry@/home/user/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:1358
Jun 28 17:42:27 gnome-session[16773]: wrapper@resource:///org/gnome/gjs/modules/lang.js:169
Jun 28 17:42:27 gnome-session[16773]: myAppIcon<.onWindowsChanged@/home/user/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:1349
Jun 28 17:42:27 gnome-session[16773]: wrapper@resource:///org/gnome/gjs/modules/lang.js:169
~~~~~~

**Comment by micheleg** [comment]
Hi, thanks for your bug report. Does this result in a crash of the extension of shell?

**Comment by hubutm20** [comment]
GnomeShell 3.14.4, i have the same issue but with "x" (can't convert "x" to an integer),and sometimes the dock stays hidden and need a GS restart to get the dock work again

**Comment by N-Coder** [comment]
For me nothing crashes at all, there are only a lot of annoying entries in the system log

**Comment by micheleg** [comment]
Does the error always appear in the log? Is there some settings or action that trigger them? I can't reproduce the problem on my system.

**Comment by hubutm20** [comment]
it could be some setting because i get another error about workspace-to-dock when i hover the mouse over the dash-to-dock "(gnome-shell:27942): Gjs-WARNING **: JS ERROR: TypeError: DashToDock.dock._animStatus is undefined
DockedWorkspaces<._onDashToDockHoverChanged@/home/hubutm20/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/dockedWorkspaces.js:773
wrapper@resource:///org/gnome/gjs/modules/lang.js:169"
this one is because "Show the dock when hovering over dash......" is activated in workspace-to-dock, this error appears in log 2 by 2: 2 when the dock appears and 2 when disappears
and after a while, appears the other error with (can't convert "x" to an integer) sometime with "y" and sometimes with other "words", i'll post them here if it is interesting what can't be converted to an integer...

**Comment by Gazedo** [comment]
I am experiencing this aswell with Gnote.
The error is:
```
Aug 03 05:26:26 ArchBox gnome-session[659]: (gnome-shell:738): Gjs-WARNING **: JS ERROR: Error: can't convert "x" to an integer
Aug 03 05:26:26 ArchBox gnome-session[659]: myAppIcon<.updateIconGeometry@/home/gazedo/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:1358
Aug 03 05:23:17 ArchBox gnome-session[659]: wrapper@resource:///org/gnome/gjs/modules/lang.js:169
Aug 03 05:23:17 ArchBox gnome-session[659]: myAppIcon<.onWindowsChanged@/home/gazedo/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/myDash.js:1349
Aug 03 05:23:17 ArchBox gnome-session[659]: wrapper@resource:///org/gnome/gjs/modules/lang.js:169
```

You might be able to replicate it by installing gnote. I'm on arch linux with gnome 3.16.2. I'm not sure if its dash-to-dock or some sort of error in gnome that dash-to-dock doesn't know how to deal with because with dash-to-dock disabled I get:
```
Aug 03 20:22:49 ArchBox gnome-session[659]: (gnome-shell:738): St-CRITICAL **: st_widget_get_theme_node called on the widget [0x31f0b40 StBoxLayout] which is not in the stage.
Aug 03 20:22:49 ArchBox gnome-session[659]: (gnome-shell:738): St-CRITICAL **: st_widget_get_theme_node called on the widget [0x7bcccc0 Gjs_DashItemContainer.dash-item-container] which is not in the stage.
Aug 03 20:22:49 ArchBox gnome-session[659]: (gnome-shell:738): St-CRITICAL **: st_widget_get_theme_node called on the widget [0x3ab23d0 StButton.app-well-app running1:last-child first-child] which is not in the stage.
```

Both of the errors repeat many times

**Comment by micheleg** [comment]
Thanks for the additional information. The second errors is clearly related to dash to dock but it looks likes due to the extension not being properly unloaded, probably due to the previous errors. In a fresh session, where DTD is never loaded, that should not happen.

**Comment by micheleg** [comment]
@Gazedo I can't reproduce the bug. Is the error 100% reproducible for you? Which action trigger the error (opening new window, menu...)? Are you working in multi-monitror configuration? Could you report your settings (you can get them with ```dconf dump /org/gnome/shell/extensions/dash-to-dock/```). Are you using a custom theme?

**Comment by chuma** [comment]
It doesn't crash anything, but it does clutter up my journal log.  I can reproduce it on my machine (Fedora 22, gnome-shell 3.16.3, extension v47) when I use an app which has multiple windows which are grouped together in the window picker, for example:

- Firefox with a Web Inspector or Firebug window open
- Terminator with multiple windows
- KeePass with a "New Entry" dialog open

In any of these situations, switching windows or locking/unlocking the screen triggers these errors.

It doesn't seem to trigger with pure GNOME apps using multiple windows.  gedit, for example, does not trigger the errors.

I'm in multi-monitor mode at the moment, not sure if this is a factor or not.  

I put some try/catches inside myAppIcon.updateIconGeometry and it appears to be choking on some rather strange values coming from get_transformed_position() and get_transformed_size(), for example:

    threw exception on position: [-9.207339222181337e+21,4.5912142885138307e-41]
    threw exception on size: [42574456,9.207704013751154e+21]
    threw exception on position: [-9.216607630214465e+21,4.5912142885138307e-41]
    threw exception on position: [4.446677908711704e-22,NaN]
    threw exception on size: [9.209347827615144e+21,1.5726691325081144e-16]
    
In all cases, one of the size or position values is large enough to be expressed in exponential notation, and none of the values seem to be useful at all.

I hope this helps... if I come up with a real solution I will be sure to send a pull request.  Still learning the ins and outs of this extension.

**Comment by micheleg** [comment]
Thank you for the additional info and investigation.

**Comment by micheleg** [comment]
Could you please try the branches bug_cant_convert_to_int_1 and bug_cant_convert_to_int_2 and see if the error is solved. I still haven't found the origin of the error because I can't reproduce it, but hopefully one of these will prevent the error.

**Comment by micheleg** [comment]
the branch bug_cant_convert_to_int_2 has been tentatively merged into master.

**Comment by chuma** [comment]
Tried out the branch bug_cant_convert_to_int_2 today - no issues, no more gnome-session warnings either.

**Comment by micheleg** [comment]
Although the reason of the bug is still not clear to me and I only managed to suppress the errors, I consider the problem solved by 8d54154a.
